### ISSUE TITLE
Feature Request: Implement an option to go to #admin room from the workspace editor

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18896]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Problem:
There is no option to go to admins room for the workspace from the workspace editor

## Solution:
`Go to # admin room` and `Go to #announce room` can be added in the 3 dot overflow

**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
<img width="960" alt="2" src="https://github.com/Expensify/App/assets/43996225/59a8c3df-149d-4c72-ad38-3f423a67bcb1">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1683935683755399

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Feature Request: Implement an option to go to #admin room from the workspace editor 
### What is the root cause of that problem?
N/A
### What changes do you think we should make in order to solve the problem?
We should add 2 new menu items here, 1. Go to `#admin` room 2. Go to `#anounce` room
https://github.com/Expensify/App/blob/2a786a71bc37004824a1012c1fd2fe4225ebf68b/src/pages/workspace/WorkspaceInitialPage.js#L147-L153

When clicked, we should something like this:
```javascript
const goTo = (type: CONST.REPORT.CHAT_TYPE.POLICY_ADMINS | CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE)=>{
        const report = _.find(props.reports, (report) => report && report.policyID === policy.id && report.chatType === type);
        Navigation.navigate(ROUTES.getReportRoute(report.reportID));
    }
``` 
which is finding a report that belongs to current policy and is a `#admin` or `#announce` room

### What alternative solutions did you explore? (Optional)

**Comment by abdulrahuman5196** [comment]
@dylanexpensify  Before approving proposals, I wanted to confirm. Are we good with adding new menu items to the workspace menus from UX perspective or Are we planning for different UX? Since it seemed to suggestion from slack thread.
 
Note: And If are good with new menu items, we would need icons and text, so it would be good to tag design team member as well here whom we can reach in case of clarifications.

**Comment by abdulrahuman5196** [comment]
@dylanexpensify  / @marcaaron  Could you kindly confirm on https://github.com/Expensify/App/issues/18896#issuecomment-1547682634 ? So that we can keep the issue moving?

**Comment by marcaaron** [comment]
>Are we good with adding new menu items to the workspace menus from UX perspective

That's what the issue is asking for so I'd say we are good with it.

>Are we planning for different/better UX? Since it seemed to be a suggestion from slack thread.

Not totally sure - can you summarize what the Slack thread says and tag in whoever is involved on that to get feedback?

>we would need icons and text, so it would be good to tag design team member as well here whom we can reach in case of clarifications.

These rooms already have default icons so I'd say we should just re-use those, but yes, curious for @shawnborton's take.

**Comment by abdulrahuman5196** [comment]
Thank you marc for the confirmation. I just wanted to confirm if we looking out for different UX just in case.

**Comment by abdulrahuman5196** [comment]
Since this issue is a straight implementation, I think this proposal - https://github.com/Expensify/App/issues/18896#issuecomment-1546731509 by @alitoshmatov is good

🎀 👀 🎀 C+ reviewed
@marcaaron

**Comment by shawnborton** [comment]
> These rooms already have default icons so I'd say we should just re-use those, but yes, curious for @shawnborton's take.

Actually I think we want to remove the room-specific icons and always use the workspace avatar. cc @MitchExpensify @grgia @JmillsExpensify not sure if this issue was ever created but it was decided during the threads predesigns, but we should implement that as soon as we can. 

As for the overflow menu, we can just reuse the hashtag room icon we use in the "New room" option in the FAB for this. And then the label would say "Go to #admins room" or something

**Comment by shawnborton** [comment]
I think it’s fine if it’s the same icon for both

**Comment by marcaaron** [comment]
LGTM @alitoshmatov just remember that not every workspace member can see the `#admins` room so that will need to be accounted for.

**Comment by alitoshmatov** [comment]
@marcaaron I thought anyone who has access to workspace page is an admin, thus can see `#admin`, am I wrong?

**Comment by marcaaron** [comment]
That sounds correct to me!

**Comment by abdulrahuman5196** [comment]
@alitoshmatov  Congratulations. Kindly let me know once the PR is ready. Would be happy to jump in and review.

Note: For translations of string kindly provide the strings here, so that we can get sorted out that as well ASAP.

**Comment by dylanexpensify** [comment]
Great work! Let's get that PR ready and get that timeline bonus! 🤑

**Comment by alitoshmatov** [comment]
@abdulrahuman5196 
PR will be ready in couple of hours
strings I am using: 
1. en - Go to #admins room
    es - Ir a la sala de #administradores
2. en - Go to #announce room 
    es - Ir a la sala de #anuncios

I just used translator for spanish, feel free to correct

**Comment by abdulrahuman5196** [comment]
@alitoshmatov  
I don't think we need to translate `#admins`, even in spanish we show the chat header as `#admins`. Just translating `Go to` should be fine in my opinion. 

Correct me if wrong @shawnborton @marcaaron

**Comment by shawnborton** [comment]
Yup, that makes sense to me.

**Comment by alitoshmatov** [comment]
PR is ready - https://github.com/Expensify/App/pull/19295

**Comment by abdulrahuman5196** [comment]
Thank you for the PR. I will review first thing in the morning.

**Comment by abdulrahuman5196** [comment]
And since every new text added and its translations has to be verified interally. Tagging @marcaaron to kindly help us with the text and translation verification. 

English = `Go to ${roomName} room`
Spanish = `Ir a la sala de ${roomName}`

**Comment by dylanexpensify** [comment]
@abdulrahuman5196 Did we get this reviewed?

**Comment by abdulrahuman5196** [comment]
@dylanexpensify Yes. Its in PR review state. Remaining action items

Translation confirmation(https://github.com/Expensify/App/issues/18896#issuecomment-1555080284) and Internal Engineer review.

**Comment by abdulrahuman5196** [comment]
@dylanexpensify If you meant for translation review, No. We are waiting on translation review.

https://github.com/Expensify/App/issues/18896#issuecomment-1555080284

It would be great if you could start the translation review process if not started already. (Not sure since its an internal process)

**Comment by dylanexpensify** [comment]
[I'm heading OOO tomorrow](https://expensify.slack.com/archives/C01SKUP7QR0/p1684776773511559) for about a week so adding another BZ member while I'm away to help keep the train moving! 🚂 Thanks @strepanier03!!

**Comment by marcaaron** [comment]
@abdulrahuman5196 Unfortunately, I do not speak español 😂 

Luckily, everyone and anyone can ask for Spanish translations in Slack! 

Here's an example you can follow: https://expensify.slack.com/archives/C01GTK53T8Q/p1641913188292300

Let us know if there are any other questions. Thanks!

**Comment by marcaaron** [comment]
I'm also going OOO starting tomorrow and will need an Engineer assigned while I'm gone. @strepanier03 would you mind helping me with this tomorrow? 🙏 

I will just need you to:

1. Temporarily unassign me
1. Re-add the `Engineering` label
1. Assign me again after a new engineer is assigned

If this one's not sorted by the time I get back I will take over the management/reviews again! Thanks!

**Comment by abdulrahuman5196** [comment]
Oh. Sorry. I thought internal engineers raise it in an internal chat room. Atleast that's how it went for my other PRs 😵‍💫 

I have raised the translation confirmation in slack again - https://expensify.slack.com/archives/C01GTK53T8Q/p1684826176793429

@alitoshmatov  Kindly do check up on that as well.

**Comment by abdulrahuman5196** [comment]
@strepanier03  C+ approval and reviewer checklist is complete. kindly involve a new engineer.

**Comment by strepanier03** [comment]
@hayata-suenaga - The previous engineer @marcaaron is out and we needed a new one, congrats on luck of the draw. 

Latest update for you:

> C+ approval and reviewer checklist is complete. kindly involve a new engineer.
> 
> PR - https://github.com/Expensify/App/pull/19295

**Comment by abdulrahuman5196** [comment]
@hayata-suenaga gentle ping on the review.

**Comment by hayata-suenaga** [comment]
ah, I missed this

@strepanier03 when you reassign an engineer, can you assign them as a reviewer to the PR? that's way, the review is displayed on my K2 dashboard's review section.

<img width="400" alt="Screenshot 2023-05-26 at 3 58 20 PM" src="https://github.com/Expensify/App/assets/98560306/0562ac70-6234-46c6-87b0-b60bad7227aa">

**Comment by hayata-suenaga** [comment]
reviewing now...

**Comment by strepanier03** [comment]
@hayata-suenaga - absolutely, thank you so much for the request, I didn't even think of that but will make sure I do from now on.

**Comment by abdulrahuman5196** [comment]
**BugZero Checklist:**

>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

New feature. Not a regression

>Determine if we should create a regression test for this bug.

Yes.

>If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

1) Go to Settings -> Workspaces -> Workspace page
2) Press three dots
3) Make sure that followings are present a)Go to #admins room b)Go to #announce room
4) Press `Go to #admins room` 
5) It should navigate to the `#admins` room of the workspace 
6) Repeat steps 1 to 3
7) Press `Go to #announce room` 
8) It should navigate to the `#announce` room of the workspace

**Comment by alitoshmatov** [comment]
Looks like this one is ready for payment

**Comment by strepanier03** [comment]
@alitoshmatov and @abdulrahuman5196 - I've hired you both the Upwork job. I'll check in again this evening to see if I can pay either out.

**Comment by strepanier03** [comment]
Added reg test request.

**Comment by strepanier03** [comment]
Looks like contracts are still waiting to be accepted so I'll check in in the morning to try to pay again.

**Comment by alitoshmatov** [comment]
@strepanier03 accepted contract in upwork

**Comment by strepanier03** [comment]
I've paid everyone, ended contracts, and left feedback. 

Thank you everyone for the hard work fixing this.
